### PR TITLE
collect: Allow configurable remote log dir

### PR DESCRIPTION
**Pull request by cawallin**
* Refactor how the port is retrieved
* Add configurable log files based on the Presto config (defaults
to the Presto default)

### CODE REVIEW

File: prestoadmin/collect.py [hunk 1]
```diff
@@ -58,20 +61,12 @@ def logs():
     """
     Gather all the server logs and presto-admin log and create a tar file.
     """
-
-    _LOGGER.debug('LOG directory to be archived: ' + REMOTE_PRESTO_LOG_DIR)
-
-    if not os.path.exists(TMP_PRESTO_DEBUG):
-        os.mkdir(TMP_PRESTO_DEBUG)
-
+    # instead, check node.server-log-file and node.launcher-log-file
```
Comment (cawallin) [inline prestoadmin/collect.py]: oops, I'll delete this comment

File: prestoadmin/util/constants.py [hunk 2]
```diff
@@ -41,5 +41,7 @@
 REMOTE_CONF_DIR = '/etc/presto'
 REMOTE_CATALOG_DIR = os.path.join(REMOTE_CONF_DIR, 'catalog')
 REMOTE_PACKAGES_PATH = '/opt/prestoadmin/packages'
-REMOTE_PRESTO_LOG_DIR = '/var/log/presto'
+DEFAULT_REMOTE_PRESTO_LOG_DIR = '/var/log/presto'
```
Comment (cawallin) [inline prestoadmin/util/constants.py]: will remove

**Comment by cawallin** [comment]
Also, I changed the behavior of log collection slightly; the directory structure for the logs is logs/hostname/server.log now, rather than logs/hostname/presto/server.log

**Comment by cawallin** [comment]
Requires RPM built with this patch to work: https://github.com/Teradata/presto/pull/61

**Comment by cawallin** [comment]
Passed all the tests in a Jenkins run.
@akshatnair you should probably look at this since you did the initial log collection stuff.

File: prestoadmin/util/remote_config_util.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        return port
+    except ConfigurationError as e:
+        raise ConfigurationError(e.message +
+                                 ' for property '
+                                 'http-server.http.port on host '
+                                 + host + '.')
+
+
+def lookup_server_log(host):
+    return lookup_string_config('node.server-log-file', NODE_CONFIG_FILE, host,
```
Comment (akshatnair) [inline prestoadmin/util/remote_config_util.py]: Does this give the directory or file? the param name is kinda confusing

**Comment by akshatnair** [comment]
lgtm mod a small comment

File: prestoadmin/util/remote_config_util.py (hunk 3 above)
Comment (cawallin) [inline prestoadmin/util/remote_config_util.py]: file, I'll update it

File: tests/unit/test_remote_config_util.py [hunk 4]
```diff
@@ -31,31 +32,65 @@ def test_lookup_port_failure(self, run_mock):
...
     def test_lookup_port_not_integer_failure(self, run_mock):
         run_mock.return_value = _AttributeString('http-server.http.port=hello')
         run_mock.return_value.failed = False
+        run_mock.return_value.return_code = 0
+
         self.assertRaisesRegexp(
             ConfigurationError,
-            'Unable to coerce http-server.http.port \'hello\' to an int. '
-            'Failed to connect to any_host.',
+            'Invalid value hello: port must be a number between 1 and 65535.'
```
Comment (ebd2) [inline tests/unit/test_remote_config_util.py]: I'd propose that this should match the out of range case and read 'Invalid port number', but if it's a pain to change, it's not worth holding this up over.

**Comment by ebd2** [comment]
LGTM.

**Comment by cawallin** [comment]
@ebd2 -- I added an explicit check (I couldn't re-use the presto-admin code b/c you have to execute it remotely). I'm pushing it now, but feel free to change it if it isn't to your liking :)

**Pull request merged**